GetResultTests failing after Jackson upgrade

Both `testToAndFromXContent` and `testToAndFromXContentEmbedded` began to fail after merging #53523.

```
org.elasticsearch.index.get.GetResultTests > testToAndFromXContent FAILED
    java.lang.AssertionError: expected:<{
      "_index" : "KwimATr",
      "_id" : "fNadN",
      "_version" : 253858921092758874,
      "_seq_no" : 5834289908670281787,
      "_primary_term" : 81,
      "_ignored" : [
        "xpw"
      ],
      "_size" : "DUAFBGG",
      "found" : true,
      "_source" : {
        "NEXaGTHjF" : -2779067364874679309,
        "wIbgXNvy" : 0.30867484486098484
      },
      "fields" : {
        "QNeadLcyz" : [
          -6659889557824123908,
          -8374625771588596860,
          -1513229210634219247
        ],
        "qkugD" : [
          1180962545832141773,
          -860339974753460802,
          1639386953175493398
        ],
        "iVaqCoaFw" : [
          1
        ],
        "xIssfiCK" : [
          "75KdXsyX0ILuir3vnoQAy4AF2Lbun7TuvoA=",
          "8r+nju6jvu61ivOaq7rvpZvno7Vy2q/kuKfwrLSvypTska3is6DgsIPxuIuK",
          "4YaxwoPutY/unoMq6LS674+t7rWrxJPvqbrum4fvq5Bg",
          "6qGTP8ez6bOs7qKTbHDnirnxqKuv",
          "w4HujI7Rtn/zj46w772R5biKzKYf7ouX7q6q4ouz7rab"
        ]
      }
    }> but was:<{
      "_index" : "KwimATr",
      "_id" : "fNadN",
      "_version" : 253858921092758874,
      "_seq_no" : 5834289908670281787,
      "_primary_term" : 81,
      "_size" : "DUAFBGG",
      "_ignored" : [
        "xpw"
      ],
      "found" : true,
      "_source" : {
        "NEXaGTHjF" : -2779067364874679309,
        "wIbgXNvy" : 0.30867484486098484
      },
      "fields" : {
        "QNeadLcyz" : [
          -6659889557824123908,
          -8374625771588596860,
          -1513229210634219247
        ],
        "qkugD" : [
          1180962545832141773,
          -860339974753460802,
          1639386953175493398
        ],
        "iVaqCoaFw" : [
          1
        ],
        "xIssfiCK" : [
          "75KdXsyX0ILuir3vnoQAy4AF2Lbun7TuvoA=",
          "8r+nju6jvu61ivOaq7rvpZvno7Vy2q/kuKfwrLSvypTska3is6DgsIPxuIuK",
          "4YaxwoPutY/unoMq6LS674+t7rWrxJPvqbrum4fvq5Bg",
          "6qGTP8ez6bOs7qKTbHDnirnxqKuv",
          "w4HujI7Rtn/zj46w772R5biKzKYf7ouX7q6q4ouz7rab"
        ]
      }
    }>
        at __randomizedtesting.SeedInfo.seed([3EAD9BBCCE2AA05:2DC37331AE9DD1F9]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.index.get.GetResultTests.testToAndFromXContent(GetResultTests.java:68)
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContent" -Dtests.seed=3EAD9BBCCE2AA05 -Dtests.security.manager=true -Dtests.locale=ar-TN -Dtests.timezone=Europe/London -Dcompiler.java=13
```

Both tests are failing with similar assertion errors.

https://gradle-enterprise.elastic.co/s/s5gd2fhhaba44/tests/nfxodyych4uas-wl5hwst2gzc56

--- a/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
@@ -136,8 +136,8 @@ public final class RandomObjects {
      */
     public static Object getExpectedParsedValue(XContentType xContentType, Object value) {
         if (value instanceof BytesArray) {
-            if (xContentType == XContentType.JSON || xContentType == XContentType.YAML) {
-                //JSON and YAML write the base64 format
+            if (xContentType == XContentType.JSON) {
+                //JSON writes base64 format
                 return Base64.getEncoder().encodeToString(((BytesArray)value).toBytesRef().bytes);
             }
         }

